veridian-slang/build.rs

```diff
@@ -1,42 +1,87 @@
 use flate2::read::GzDecoder;
 use std::env;
-use std::fs::File;
-use std::path::PathBuf;
+use std::fs::{self, File};
+use std::path::{Path, PathBuf};
 use tar::Archive;
 
-fn download_slang() -> Result<(), Box<dyn std::error::Error>> {
-    let target = "https://github.com/MikePopoloski/slang/releases/download/v0.7/slang-linux.tar.gz";
-    let fname = "slang-linux.tar.gz";
-    let mut response = reqwest::blocking::get(target)?;
-    let mut dest = File::create(fname)?;
-    response.copy_to(&mut dest)?;
-    let tar = GzDecoder::new(File::open(fname)?);
-    let mut archive = Archive::new(tar);
-    archive.unpack("slang_wrapper/.")?;
-    Ok(())
+type Result<T> = std::result::Result<T, Box<dyn std::error::Error>>;
+
+fn download_slang(download_to: &Path) -> Result<PathBuf> {
+    // Keep the version the same as the one in `CMakeLists.txt`
+    let target = "https://github.com/MikePopoloski/slang/archive/refs/tags/v6.0.tar.gz";
+
+    fs::create_dir_all(download_to)?;
+
+    // Download source
+    let archive_path = download_to.join("slang-linux.tar.gz");
+    let mut dest = File::create(&archive_path)?;
+    reqwest::blocking::get(target)?.copy_to(&mut dest)?;
+    drop(dest);
+
+    // Unpack archive
+    let mut archive = Archive::new(GzDecoder::new(File::open(archive_path)?));
+    archive.unpack(download_to)?;
+
+    // Return the source directory
+    let entries = fs::read_dir(&download_to)?
+        .filter_map(|entry| entry.ok())
+        .filter(|entry| entry.metadata().is_ok_and(|e| e.is_dir()))
+        .collect::<Vec<_>>();
+    // Expected exactly one directory in archive
+    assert_eq!(entries.len(), 1);
+    Ok(entries.first().unwrap().path())
 }
 
-fn build_slang_wrapper() {
-    cc::Build::new()
-        .cpp(true)
-        .flag("-std=c++17")
-        .flag("-Wno-type-limits")
-        .static_flag(true)
-        .include("slang_wrapper/slang/include")
-        .file("slang_wrapper/src/slang_lib.cpp")
-        .file("slang_wrapper/src/basic_client.cpp")
-        .out_dir("slang_wrapper/slang/lib")
-        .compile("slangwrapper");
+fn build_slang(slang_src: &Path, slang_install: &Path) {
+    cmake::Config::new(slang_src)
+        .profile("Release")
+        .out_dir(slang_install)
```

Comment:
```suggestion
        .define("SLANG_USE_MIMALLOC", "OFF")
        .out_dir(slang_install)
```
This change allows slang to build, since it seems to be unable to link to the dynamic library for mimalloc

Reply:
Done.